trin-core/src/utils.rs

```diff
@@ -16,6 +16,30 @@ pub fn xor_two_values(first: &[u8], second: &[u8]) -> Vec<u8> {
         .collect()
 }
 
+pub fn get_data_dir(local_enr: Enr) -> String {
+    let path = env::var(TRIN_DATA_ENV_VAR).unwrap_or_else(|_| get_default_data_dir(local_enr));
+
+    fs::create_dir_all(&path).expect("Unable to create data directory folder");
+    path
+}
+
+pub fn get_default_data_dir(local_enr: Enr) -> String {
+    // Windows: C:\Users\Username\AppData\Roaming\Trin\data
+    // macOS: ~/Library/Application Support/Trin
+    // Unix-like: $HOME/.local/share/trin
+
+    // Append last 8 enr base64 encoded chars to application dir name
+    let mut application_string = "Trin_".to_owned();
+    let len = &local_enr.to_base64().len();
+    let suffix = &local_enr.to_base64()[len - 8..];
+    application_string.push_str(suffix);
+
+    match ProjectDirs::from("", "", &application_string) {
+        Some(proj_dirs) => proj_dirs.data_local_dir().to_str().unwrap().to_string(),
+        None => panic!("Unable to find data directory"),
+    }
+}
+
```

Comment:
Why was this moved? Makes it tougher to see what was changed.

Reply:
Imo - tests belong at the bottom of the file. Not sure if that's convention or not in Rust? Happy to switch it back tho

Reply:
Ah yes, I missed that the tests were basically what moved. 👍🏻 I don't know the standards that well either, but tests at the bottom is the only way I've seen it, too.